app/chaturbate/save.go

```diff
@@ -16,26 +19,34 @@ type saveData struct {
 }
 
 type saveLog struct {
+	Mes string
 	Rid int64
 	Now int64
-	Mes string
 }
 
-func getDonId(name string) int64 {
+func getDonId(name string) (int64, bool) {
 	donator := new(tID)
 	err := Mysql.Get(donator, "SELECT id FROM donator WHERE name=?", name)
 	if err != nil {
-		res, _ := Mysql.Exec("INSERT INTO donator (`name`) VALUES (?)", name)
-		donator.Id, _ = res.LastInsertId()
+		if err == sql.ErrNoRows { // ошибка может быть разная, если мы ничего не нашли всегда возвращается sql.ErrNoRows, при других ошибках вернется иное, лучше делать разделение логики в этом случае
```

Comment:
database errors may contains ErrNoRows if no rows found and permament db error like broken connection or database mailfunction (like no free space or other errors...)